### PR TITLE
Chore: add plugin code

### PR DESCRIPTION
**Pull request by HeverFarber**
This env0 plugin will allow you to trigger only terragrunt modules that were changed in the current Pull Request.

Solution:
* Use zx to run javascript as bash. 
* Run diff, remove working directory, unique, inject to ENV VAR

### CODE REVIEW

File: modules-detection.mjs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+await detect();
+
+async function detect() {
+    if (!sourceRef || !targetRef) {
+        console.log(`Skipping - can't detect source or target refs. source [${sourceRef}] target [${targetRef}]`);
+        return;
+    }
+
+    const { stdout: fileChangesRaw } = await $`git diff --name-only  ${sourceRef}..${targetRef}`;
```
Comment (alonnoga) [inline modules-detection.mjs:17]: you installed that lib in our image? ( the one enabling you to $`command` ?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,20 @@
-# env0-terragrunt-modules-detection-plugin
-The plugin allows you to run terragrunt run-all only on the modules who change between commits
+# env0 terragrunt modules detection
+This env0 plugin will allow you to trigger only terragrunt modules that were changed in the current Pull Request.
+
+**Currently, the plugin only support [Pr Plan](https://docs.env0.com/docs/plan-on-pull-request) and it will detect all modules change for specific PR.**
+
+## Inputs
+The plugin doesn't support any inputs.
```
Comment (alonnoga) [inline README.md]: a bit less negative
```suggestion
The plugin doesn't require any inputs.
```

**Review by alonnoga — APPROVED** [review]
LGTM

File: modules-detection.mjs (hunk 1 above)
Comment (HeverFarber) [reply modules-detection.mjs:17]: Nope. I am installeing it part of the plugin https://github.com/env0/env0-terragrunt-modules-detection-plugin/blob/8366ea8c05362b615fa8500162858f1abeb6fdd7/env0.plugin.yml#L5

File: README.md (hunk 2 above)
Comment (HeverFarber) [reply README.md]: 35090eb

**Pull request merged**